tests/kafkatest/services/verifiable_producer.py

```diff
@@ -87,10 +109,10 @@ def _worker(self, idx, node):
                 with self.lock:
                     if data["name"] == "producer_send_error":
                         data["node"] = idx
-                        self.not_acked_values.append(int(data["value"]))
+                        self.not_acked_values.append(self.message_validator(data["value"]))
```

Comment:
Now that `message_validator` methods are being used here in VerifiableProducer, it is odd that with the existing validators we the behavior is now:
If valid, append original value
else, append None

That actually seems pretty surprising. Previously, if the value was not parseable, an error would be thrown immediately.